guide/results-viewer-react/src/components/YamlUrls/RequestBodyView.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+
+export interface RequestBodyViewProps {
+    requestBody: object | undefined;
+    updateRequestBody: (newRequestBody: object) => void;
+}
+
+function RequestBodyView ({ requestBody, updateRequestBody }: RequestBodyViewProps): JSX.Element {
+    const [requestBodyInEdit, setRequestBodyInEdit] = React.useState<string>(JSON.stringify(requestBody, null, 2).replace(/\\n/g, "\n").replace(/^"|"$/g, ""));
+
+    const requestBodyDisplayStyle: React.CSSProperties = {
+        fontFamily: "monospace",
+        width: "100%",
+        height: "270px",
+        overflow: "auto",
+        border: "1px solid #ccc",
+        padding: "10px",
+        backgroundColor: "#2e3438",
+        resize: "none",
+        tabSize: 2,
+        boxSizing: "border-box"
+    };
+
+    const handleEditorInput = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+        const newRequestBody = event.target.value;
+        setRequestBodyInEdit(newRequestBody);
+        try {
+            updateRequestBody(JSON.parse(newRequestBody));
+        } catch (error) {
+            console.error(error);
```

Comment:
There should be a logger you can pull in from util

Reply:
I'm going to be making the errors more robust for this view. That will replace this console.error when I implement it. Also, I'm unfamiliar with how to use that logger. Are you amenable to leave this as is for now? I have a story on the backlog to fix the error logging.

Reply:
Ah, now I see it's causing a build error. If the logger is the best way to go, can you point me in the direction of an example?

Reply:
Here's an example of the logger in use. It does a console.log (or error) behind the scenes but lets you add more data to explain rather than just spitting the error into the console.

https://github.com/FamilySearch/pewpew/blob/master/guide/results-viewer-react/src/components/TestResults/index.tsx#L196